src/com/sap/piper/GitUtils.groovy

```diff
@@ -4,6 +4,11 @@ boolean insideWorkTree() {
     return sh(returnStatus: true, script: 'git rev-parse --is-inside-work-tree 1>/dev/null 2>&1') == 0
 }
 
+boolean isWorkTreeClean() {
+    if(!insideWorkTree()) error 'Method \'isWorkTreeClean\' called outside a git work tree.'
```

Comment:
In the only use case of this function is is already clear that we are in a workspace, remove one of the checks.